El diagrama de la figura corresponde a un receptor super-heterodino de simple conversión con $F_I = 10,7 MHz$ y sintonizado para recibir una señal de $F_{RF} = 100 MHz$

![Parcial](fig/parcial20200824aa.png) 

Mezclador:  $NF_{mix}= 10dB$ y $G_{mix} = 0 dB$

Amplificador Frecuencia intermedia:  $NF_{IF}  = 6 dB$  y  $G_{IF}= 40 dB$

![Parcial](fig/parcial20200824a.png) 

El circuito de la figura corresponde a la etapa de RF (filtro y  amplificador), tiene configuración Butterworth con un ancho de banda $BW=1MHz$. 

Todas las interfaces están adaptadas a MTE Qcte y los elementos reactivos no tienen pérdidas.

El amplificador tiene configuración Butterworth y ancho de banda $BW=1MHz$

Todas las interfaces están adaptadas a MTE y los elementos reactivos no tienen pérdidas


$Z_{antena}= (50+j0)  \Omega$

$T_{antena}= 400\,K$

$R_L = 1 K\Omega$		

![Parcial](fig/parcial20200824c.png) 

$Y_{11g1}= (10 + j0) \mu S$

$Y_{22}  = (50 + j0) \mu S$

$Y_{11g2}= Y_{12g1} =Y_{12g2} = 0$		

$mod(Y_{21g1})= g_m$     

$ang(Y_{21g1})=-30 grados$

Determinar:
Siendo el DS subacoplado el de la entrada y $V_{g2}= 5Vdc$
1. (1 punto) Q1, h1 correspondientes al DS de entrada                             
2. (1 punto) Q2, h2 correspondientes al DS de salida                               
3. (1 punto) Rechazo de frecuencia imagen del receptor                
4. (1 punto) Valor de los componentes del circuito de RF 
5. (1 punto) $V_o(100\,MHz)$ considerando que $V_{ant}= 10 \mu V$            
6. (1 punto) Considerando que el ancho de banda equivalente de ruido de la etapa de IF es de $BEN=100\,KHz$, determinar el valor de tensión de la señal en antena cuando a la salida del IF hay una $SNR_o = 20\,dB$

Se elimina el DS subacoplado y se lo reemplaza por un acoplador aperiódico que mantiene la adaptación de impedancias en banda ancha ($V_{g2}= 5Vdc$)
7. (1 punto) Determinar el nuevo ancho de banda               
8. (1 punto) Determinar el riple en banda de paso.   

Se reemplazan los DS de entrada y salida por SS de igual Q que el DS reemplazado manteniéndose las condiciones de adaptación a MTE.
9. (2 puntos) Considerando que $Y_{12g1}$ está dada por una capacidad de $C=0.01\,pf$, determinar el máximo valor de $V_{g2}$ que permite que el circuito sea estable.                                        


In [2]:
import numpy as np

# Parametros del transistor 
r11   =   1/10e-6 
r22   =   1/50e-6 
gm    = 10e-3 
rg    =   50
rgate = 10e3
rl    =   1e3
Qo = np.inf

# Calculos auxiliares
####### 1
BW = 1e6
fo = 100e6
wo = (2*np.pi*fo)



fo = 100e6
BW = 1e6
n  = 4

##  1. (1 punto) Q1, h1 correspondientes al DS de entrada      

La figura ilustra la ubicación de los polos de los dobles sintonizados. 

![Polos](fig/butterdosdobles_20200824.png)

De la figura, los polos correspondientes al primer doble sintonizado permiten calcular $Q$,$k$ y $h$ en base a su ubicación. 

$f_o = 100 MHz$

$BW = 1 MHz$

$Q_1  = \frac{f_o}{2 \cdot \frac{BW}{2} \cdot sin(45 + \frac{45}{2})} = 108.23$

$K_1  = \frac{BW \cdot cos(45+\frac{45}{2})}{fo}  = 0.0038$

$h_1  = Q_2 \cdot  K_2  = 0.41$

In [3]:

Q1 = fo/(BW*np.sin((np.pi/n+np.pi/(2*n))))
K1 = (BW*np.cos((np.pi/n+np.pi/(2*n))))/fo
h1 = K1*Q1
print('$$Q1 =  {:1.2f}$$'.format(Q1))
print('$$K1 =  {:1.4f}$$'.format(K1))
print('$$h1 =  {:1.2f}$$'.format(h1))

$$Q1 =  108.24$$
$$K1 =  0.0038$$
$$h1 =  0.41$$


## 2. (1 punto) Q2, h2 correspondientes al DS de salida   

De igual manera para el segundo doble sintonizado:

$Q_2  = \frac{f_o}{2 \cdot \frac{BW}{2} \cdot sin(\frac{45}{2})} = 261.31$

$K_2  = \frac{BW \cdot cos(\frac{45}{2})}{fo}  = 0.0092$

$h_1  = Q_2 \cdot  K_2  = 0.41$

In [4]:
Q2 = fo/(BW*np.sin(np.pi/(n*2)) )
K2 = BW*np.cos(np.pi/(n*2))/fo
h2 = K2*Q2

print('$$Q2 =  {:1.2f}$$'.format(Q2))
print('$$K2 =  {:1.4f}$$'.format(K2))
print('$$h2 =  {:1.4f}$$'.format(h2))

$$Q2 =  261.31$$
$$K2 =  0.0092$$
$$h2 =  2.4142$$


## 3. (1 punto) Rechazo de frecuencia imagen del receptor                

Usando la transferencia del filtro Butterworth:

$|A| = \frac{1}{\sqrt{1 +  (\frac{f_{imag} - f_o}{f_c - f_o}})^{2\cdot4} }$

$|A| = \frac{1}{\sqrt{1 +  (\frac{2 \cdot 10.7MHz}{\frac{1MHz}{2}}})^{8} } = $




In [5]:
fcs  = 100e6
BW   =   1e6
fif  =   10.7e6
fimg = fcs + 2* fif 
n    = 4 


Avimg = 1/(1+( (2*fif)/(BW/2))**(2*n) )**0.5

print('$$|  A_v(img) | = {:1.2e}$$'.format(Avimg) )

print('$$|  A_v(img) | = {:1.2f}dB$$'.format(20*np.log10(Avimg)) )

$$|  A_v(img) | = 2.98e-07$$
$$|  A_v(img) | = -130.52dB$$


## 4. (1 punto) Valor de los componentes del circuito de RF 

Primero doble sintonizado:


$$L_1 = \frac{r_g}{\omega_o Q_1}$$


$$C_1 = \frac{1}{\omega_o^2 L_1}$$

$$r_i = \frac{r_{11} r_{g}}{r_{11} + r_{g}}$$ 

$$L_2 = \frac{r_{i}}{\omega_o Q_1}$$

$$C_2 = \frac{1}{\omega_o^2 L_2}$$

Segundo Doble sintonizado

$$L_3 = \frac{r_22}{\omega_o Q_2}$$

$$C_3 = \frac{1}{\omega_o^2 L_3}$$


$$L_4 = \frac{r_{l}}{\omega_o Q_2}$$

$$C_4 = \frac{1}{\omega_o^2 L_4}$$




In [6]:


L1 = (1/Q1-1/Qo)*rg/wo
C1 = 1/(L1*wo**2) 

L2 = (1/Q1-1/Qo)*(r11*rgate/(r11+rgate))/wo
C2 = 1/(L2*wo**2) 

print('$$L1= {:1.2e} Hy$$'.format(L1))
print('$$C1= {:1.2e} F$$'.format(C1))

print('$$L2= {:1.2e} Hy$$'.format(L2))
print('$$C2= {:1.2e} F$$'.format(C2))


L3 = (1/Q2-1/Qo)*r22/wo
C3 = 1/(L3*wo**2) 

L4 = (1/Q2-1/Qo)*rl/wo
C4 = 1/(L4*wo**2) 

print('$$L3= {:1.2e} Hy$$'.format(L3))
print('$$C3= {:1.2e} F$$'.format(C3))

print('$$L4= {:1.2e} Hy$$'.format(L4))
print('$$C4= {:1.2e} F$$'.format(C4))

$$L1= 7.35e-10 Hy$$
$$C1= 3.45e-09 F$$
$$L2= 1.34e-07 Hy$$
$$C2= 1.89e-11 F$$
$$L3= 1.22e-07 Hy$$
$$C3= 2.08e-11 F$$
$$L4= 6.09e-09 Hy$$
$$C4= 4.16e-10 F$$


## 5. (1 punto) $V_o(100\,MHz)$ considerando que $V_{ant}= 10 \mu V$            


La trasferencia del primero doble sintonizado:

$$\frac{v_1}{i_g}= j \cdot \sqrt{L_1 L_2} \cdot  \omega_o \cdot Q_1 \frac{h_1}{ [1+j (\chi-h_1))] [1+j (\chi+h_1))]}$$

$$v_i = i_g \cdot r_g$$


$$i_g = \frac{v_i}{r_g}$$


$$\frac{v_1}{v_i}= j \cdot \frac{1}{r_g} \sqrt{L_1 L_2} \cdot  \omega_o \cdot Q_1 \frac{h_1}{ [1+j (\chi-h_1))] [1+j (\chi+h_1))]}$$


La transferencia del segundo doble sintonizado:

$$\frac{v_o}{v_1}= j \cdot g_m \cdot \sqrt{L_3 L_4} \cdot  \omega_o \cdot Q_2 \frac{h_2}{ [1+j (\chi-h_2))] [1+j (\chi+h_2))]}$$



$$|\frac{v_o}{v_i}(\chi_o)|= |\frac{v_1}{v_i}(\chi_o)| \cdot |\frac{v_o}{v_1}(\chi_o)|$$ 




In [7]:
v1ig = (1/rg)*(L1*L2)**0.5*wo*Q1*(h1/(1+h1**2))
print('$$Av1  = {:1.3f}$$'.format(v1ig))

vov1 = gm*(L3*L4)**0.5*wo*Q2*(h2/(1+h2**2))
print('$$Av2  = {:1.3f}$$'.format(vov1))

Av = v1ig*vov1
print('$$Av  = {:1.3f}$$'.format(Av))

$$Av1  = 4.767$$
$$Av2  = 15.811$$
$$Av  = 75.378$$


## 6. (1 punto) Considerando que el ancho de banda equivalente de ruido de la etapa de IF es de $BEN=100\,KHz$, determinar el valor de tensión de la señal en antena cuando a la salida del IF hay una $SNR_o = 20\,dB$

$G_{rf} = |A_v|^2 \frac{r_i}{r_o}$

Amplificador RF: $NF_{rf}= 3 dB$ y $G_{rf} =  dB$

Mezclador:  $NF_{mix}= 10dB$ y $G_{mix} = 0 dB$

Amplificador Frecuencia intermedia:  $NF_{IF}  = 6 dB$  y  $G_{IF}= 40 dB$

### Cifra de ruido

$F_t = F_{RF} + \frac{F_{mix1}-1}{G_{RF}} + \frac{F_{IF1}-1}{G_{RF} G_{mix1}} + \frac{F_{mix2}-1}{G_{RF}G_{mix1}G_{IF1}}$ 



In [8]:
#Grf  = 25

Grf = (Av**2 * rg/rl)

NFrf  = 3 

Gmix1d = 0
NFmix1= 10

Gif1d  = 40
NFif1 = 6


print('$$Grf = {:1.2f}$$'.format(Grf))
print('$$Grf = {:1.2f}dB$$'.format(10*np.log10(Grf) ))
Frf   = 10**(NFrf /10)

Frf   = 10**(NFrf  /10)
Fmix1 = 10**(NFmix1 /10)
Fif1  = 10**(NFif1  /10)


Gmix1 = 10**(Gmix1d /10)
Gif1  = 10**(Gif1d  /10)


Ft = Frf + (Fmix1 -1)/(Grf) + (Fif1 -1)/(Grf*Gmix1)

print('$$Ft  = {:1.2f}$$'.format(Ft))
print('$$NFt = {:1.2f}dB$$'.format(10*np.log10(Ft) ))


Tant = 400
BEN  = 100e3

SNRod = 20

SNRo = 10**(SNRod/10)

SNRi = Ft * SNRo

Ni    = 1.38e-23 * Tant * BEN
print(Ni)
Si    = SNRi * Ni
print(Si)
Vi    = (Si*rg)**0.5 
print('$$Vi = {:1.2e}$$'.format(Vi))

$$Grf = 284.09$$
$$Grf = 24.53dB$$
$$Ft  = 2.04$$
$$NFt = 3.09dB$$
5.52e-16
1.1246644994295441e-13
$$Vi = 2.37e-06$$




## 7. (1 punto) Se elimina el DS subacoplado y se lo reemplaza por un acoplador aperiódico que mantiene la adaptación de impedancias en banda ancha ($V_{g2}= 5Vdc$): determinar el nuevo ancho de banda               

$A(\chi) = \frac{j gm M w_o Q^2  }{(1+j(\chi+h))(1+j(\chi-h)) }$ 

$A(\chi_o) = \frac{j gm M w_o Q^2  }{(1+j(h))(1+j(h)) } = \frac{j gm M w_o Q^2  }{(1+h^2) }$ 

$\bar{A}(\chi) = \frac{A(\chi)}{A(\chi_o)} = \frac{1+h^2 }{ (1+j(\chi+h))(1+j(\chi-h)) }$ 


$| \bar{A}(\chi) | =  \frac{1+h^2 }{\sqrt{\chi^4+2\chi^2(1-h^2)+(1+h^2)^2} } = \frac{1}{\sqrt{2}}$ 


$| \bar{A}(\chi) | =  \frac{(1+h^2)^2 }{\chi^4+2\chi^2(1-h^2)+(1+h^2)^2}  = \frac{1}{2}$ 

$\chi^4+2\chi^2(1-h^2)+(1+h^2)^2 = 2 \cdot (1+h^2)^2$

$\chi^4+2\chi^2(1-h^2) =  (1+h^2)^2$

$\chi^4+2\chi^2(1-h^2)   - (1+h^2)^2 = 0$


$x = \frac{-b + \sqrt{b^2- 4 \cdot a \cdot c}}{2 \cdot a}$

$\chi = \sqrt{x}$

$\frac{BW}{2} = \frac{\chi \cdot fo }{2 \cdot Q}$


In [9]:
def Solve_cuad(a,b,c):
    d = b**2-4*a*c
    x1 = (-b+ (b**2-4*a*c)**0.5 )/(2*a)
    x2 = (-b+ (b**2+4*a*c)**0.5 )/(2*a)
    return x1 ,x2


c = -1*(1+h2**2)**2
b =  2*(1-h2**2)
a =  1

x1, x2 = Solve_cuad(a,b,c)


print('X1 =  {:1.2f}'.format(x1))
print('X2 =  {:1.2f}'.format(x2))

X  = (  (x1))**0.5
print(X) 

BW = X*fo/(Q2)
print('BW =  {:1.2f} MHz'.format(BW/1e6))


X1 =  13.19
X2 =  nan
3.632011594894805
BW =  1.39 MHz


/home/aalmela/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: RuntimeWarning: invalid value encountered in double_scalars
  after removing the cwd from sys.path.


## 8. (1 punto) Determinar el riple en banda de paso.   

$A(\chi) = \frac{j gm M w_o Q^2  }{(1+j(\chi+h))(1+j(\chi-h)) }$ 

$A(\chi_o) = \frac{j gm M w_o Q^2  }{(1+j(h))(1+j(h)) } = \frac{j gm M w_o Q^2  }{(1+h^2) }$ 

$\bar{A}(\chi) = \frac{A(\chi)}{A(\chi_o)} = \frac{1+h^2 }{ (1+j(\chi+h))(1+j(\chi-h)) }$ 


$| \bar{A}(\chi) | =  \frac{1+h^2 }{\sqrt{\chi^4+2\chi^2(1-h^2)+(1+h^2)^2} } = \frac{1}{\sqrt{2}}$ 


$| \bar{A}(\chi) | =  \frac{(1+h^2)^2 }{\chi^4+2\chi^2(1-h^2)+(1+h^2)^2}  = \frac{1}{2}$ 

Buscamos los máximo y minimos de la transferencia:

$\frac{d}{d\chi} (\chi^4+2\chi^2(1-h^2)+(1+h^2)^2)= 0$ 


$4 \chi^3 + 4\chi(1-h^2) = 0$ 

$\chi = 0$

$4 \chi^2 = - 4 (1-h^2)$ 

$\chi^2  = h^2 - 1$ 

$\chi^2  = h^2 - 1 =  4.8$ 

$\chi   = 2.192$ 

$| \bar{A}(\chi) | =  \frac{(1+h^2)^2 }{\chi^4+2\chi^2(1-h^2)+(1+h^2)^2}$

Tranferencia $\chi = 0$:  

$| \bar{A}(\chi= 0) | =  \frac{1+(2.41)^2 }{\sqrt{0^4+2\cdot 0^2 \cdot (1-(2.41)^2)+(1+(2.41)^2)^2} }$ 

$| \bar{A}(\chi= 0) | =  \frac{1+(2.41)^2 }{(1+(2.41)^2)} = 1$ 

Tranferencia $\chi = 2.192$:  

$| \bar{A}(\chi= 0) | =  \frac{1+(2.41)^2 }{\sqrt{2.192^4+2\cdot 2.192^2 \cdot (1-(2.41)^2)+(1+(2.41)^2)^2} } = 1.412$ 




In [10]:
def ADS(X,h):
    return (1+h**2)/(X**4+2*X**2*(1-h**2)+(1+h**2)**2)**0.5


Xmin = 0 

Xmax = (h2**2-1)**0.5

print('Xmin =  {:1.2e}'.format(Xmin))

print('Xmax =  {:1.2e}'.format(Xmax))

Amin = ADS(Xmin,h2)

Amax = ADS(Xmax,h2)

print('Amin =  {:1.2e}'.format(Amin))

print('Amax =  {:1.2e}'.format(Amax))

Xmin =  0.00e+00
Xmax =  2.20e+00
Amin =  1.00e+00
Amax =  1.41e+00


## 9. (2 puntos) Considerando que $Y_{12g1}$ está dada por una capacidad de $C=0.01\,pf$, determinar el máximo valor de $V_{g2}$ que permite que el circuito sea estable.                                        


$$|y_{12} y_{21}|+\Re{(y_{12} y_{21})}$$

$$|y_{12} y_{21}| = |y_{12}| |y_{21}|$$ 


$$\Re{(y_{12} y_{21})} = \Re{((g_{21}+j b_{21}) (j b_{12})}$$

$$\Re{((g_{21}+j b_{21}) (j b_{21})} =- b_{21} b_{12}$$

$$b_{21} = gm sen(\theta)$$

$$\Re{((g_{21}+j b_{21}) (j b_{21})} = - gm sen(\theta) b_{12}$$

$$|y_{12} y_{21}|+\Re{(y_{12} y_{21})} = gm b_{12} - gm sen(\theta) b_{12}$$

$$|y_{12} y_{21}|+\Re{(y_{12} y_{21})} = gm (b_{12} - sen(\theta) b_{12})$$


In [11]:
def gmTranf(gm):
    if gm <= 1e-3: Vo = gm/1e-3
    else         : Vo = (gm - 1e-3)*4/9e-3 + 1 
    return Vo

fo = 100e6
wo = 2*np.pi*fo
c = 0.01e-12
# Parametros del transistor 100 MHz
y11 =   10e-6 
y12 =   1j*wo*c
y22 =   50e-6 
y21th = -30*np.pi/180

##y21 =  ?? 
#K = (2*(y11.real)*(y22.real)) / (abs(y12*y21)+(y12*y21).real)


gm = (2*(y11.real*2)*(y22.real*2)) / (abs(y12)* (1-np.sin(y21th)) )
print( (abs(y12)* (1-np.sin(y21th)) ))

print(gm)



print(gmTranf(gm))

9.424777960769379e-06
0.00042441318157838764
0.42441318157838764


In [12]:
import numpy as np

# Parametros del transistor 
r11   =   1/10e-6 
r22   =   1/50e-6 
gm    = 10e-3 
rg    =   50
rgate = 10e3
rl    =   1e3
Qo = np.inf

# Calculos auxiliares
####### 1
BW = 1e6
fo = 100e6
wo = (2*np.pi*fo)

# Doble entrada 
ang = 360/8

Q2 = fo/(BW*np.sin((ang/2)*np.pi/180))

k2 = (BW*np.cos((ang/2)*np.pi/180))/fo

h2 = Q2 * k2


Q1 = fo/(BW*np.sin((ang*3/2)*np.pi/180))

k1 = (BW*np.cos((ang*3/2)*np.pi/180))/fo

h1 = k1 * Q1

print('$$Q_1= {:1.3f}$$'.format(Q1))

print('$$k_1 = {:1.3f}$$'.format(k1))

print('$$h_1 = {:1.3f}$$'.format(h1))



print('$$Q_2= {:1.3f}$$'.format(Q2))

print('$$k_2 = {:1.3f}$$'.format(k2))

print('$$h_2 = {:1.3f}$$'.format(h2))



$$Q_1= 108.239$$
$$k_1 = 0.004$$
$$h_1 = 0.414$$
$$Q_2= 261.313$$
$$k_2 = 0.009$$
$$h_2 = 2.414$$


In [13]:
v1ig = (1/rg)*(L1*L2)**0.5*wo*Q1*(h1/(1+h1**2))
print('$$Av1  = {:1.3f}$$'.format(v1ig))

vov1 = gm*(L3*L4)**0.5*wo*Q2*(h2/(1+h2**2))
print('$$Av2  = {:1.3f}$$'.format(vov1))

Av = v1ig*vov1
print('$$Av  = {:1.3f}$$'.format(Av))

$$Av1  = 4.767$$
$$Av2  = 15.811$$
$$Av  = 75.378$$
